### ISSUE TITLE
Unable to run on a real IOS device

### ISSUE THREAD

**Issue by chris1248** [appium/ruby_lib#342]
problem:

I am unable to get a script, running this poorly documented appium lib, to connect through appium to a real IOS device.

background:
[mac os x Yosemite]
[Appium 1.3.7]

I can successfully run the following script on an IOS simulator:

    require 'appium_lib'
    desired_caps = {
      caps:       {
        platformName: 'iOS',
        platformVersion: '8.3',
        deviceName:    'iPhone 6 Plus',
        browserName:   'Safari',
          }
    }

    # Start the driver
    driver = Appium::Driver.new(desired_caps)
    driver = driver.start_driver

    driver.get("http://www.google.com")

    element = driver.find_element(:name, 'q')
    element.send_keys "Selenium Stuff"
    element.submit
    sleep 5
    driver.quit

But I want to get it to run on a **real** device.

When I start up Appium (with safarilauncher), and connected to my IPad Air, I get this exception:

**.rbenv/versions/2.1.5/lib/ruby/gems/2.1.0/gems/appium_lib-6.0.0/lib/appium_lib/driver.rb:485:in `rescue in start_driver': ERROR: Unable to connect to Appium. Is the server running? (RuntimeError)`**

For Appium, which I'm running using the GUI, I have the following settings:

Launching Appium with command: '/Applications/Appium.app/Contents/Resources/node/bin/node' lib/server/main.js --address "10.0.20.137" --command-timeout "7200" --session-override --pre-launch --platform-version "8.3" --platform-name "iOS" --app "com.bridgemobile.testing" --udid "4d18922f629a81be51e746619cb33f81570f9572" --backend-retries "3" --default-device --launch-timeout "90000"

So of course my server is running, I can see in Appium that it spit out tons of information messages in the appium window and such. But this ruby library seems to be not be able to find it.

Why?

**Comment by JaniJegoroff** [comment]
Have you looked at this issue: https://github.com/appium/appium/issues/5100

**Comment by chris1248** [comment]
I have now. I didn't see anything new there.

**Comment by chris1248** [comment]
I have now, even pulled down the appium source code from git. I built it:
**./reset --ios --real-safari**
and ran appium from the command line (without the GUI). Appium started up just fine, and gave me the standard normal startup information to the command shell. Ok great. 
I started up the webkit proxy thingy too:
**./bin/ios-webkit-debug-proxy-launcher.js -c 4d18922f629a81be51e746619cb33f81570f9572:27753 -d**

I then fire up node with the following args: 
**node lib/server/main.js --command-timeout "7200" --session-override --pre-launch --platform-version "8.3" --platform-name "iOS" --app "com.bridgemobile.teing" --udid "4d18922f629a81be51e746619cb33f81570f9572" --log-level debug**

Last of all I execute my test. It successfully fires up safari launcher (which just sits there like a dead duck), and nothing happens on the ipad. meanwhile on the command line, node is spitting this out in response to the command to browser to google:

`info: --> POST /wd/hub/session/a6533466-cd57-4e06-992a-020075b23998/url {"url":"http://www.google.com"}
info: [debug] Responding to client with success: {"status":0,"value":"","sessionId":"a6533466-cd57-4e06-992a-020075b23998"}
info: <-- POST /wd/hub/session/a6533466-cd57-4e06-992a-020075b23998/url 200 0.942 ms - 74 {"status":0,"value":"","sessionId":"a6533466-cd57-4e06-992a-020075b23998"}
info: --> POST /wd/hub/session/a6533466-cd57-4e06-992a-020075b23998/element {"using":"name","value":"q"}
warn: [DEPRECATED] The name locator strategy has been deprecated and will be removed.  Please use the accessibility id locator strategy instead.
info: [debug] Waiting up to 30000ms for condition
info: [debug] Pushing command to appium work queue: "au.getElementByName('q')"
info: [debug] Sending command to instruments: au.getElementByName('q')`

**Comment by chris1248** [comment]
decided to not use the SafariLauncher I had previously built and put on my ipad. But instead I wanted to use the safarilauncher that came with the source code, and was built by the ./reset.sh script. 
So I took out the argument for --app, and specified --safari instead. Here is my command line:

node lib/server/main.js --command-timeout "7200" --session-override --pre-launch --platform-version "8.3" --platform-name "iOS" --udid "4d18922f629a81be51e746619cb33f81570f9572"  --safari

Once I did that, I was **successful** in running my script on the real device. \o/
